### PR TITLE
Fix Change NavigationStack as DataMember

### LINKED ISSUE #3322: [Bug]: Router State is No Longer Saved To Disk
### Describe the bug 🐞

Earlier, we were able to save `RoutingState.NavigationStack` to disk, as the collection was annotated as a `[DataMember]`: https://github.com/reactiveui/ReactiveUI/blame/a43e99ae2fbcd644460f245f471f573b7d3b66ec/src/ReactiveUI/Routing/RoutingState.cs#L26-L27

But now, the observable collection is `[IgnoreDataMember]`d: https://github.com/reactiveui/ReactiveUI/blob/main/src/ReactiveUI/Routing/RoutingState.cs#L50-L51

### Step to reproduce

1. Clone https://github.com/reactiveui/ReactiveUI.Samples/
2. Navigate to https://github.com/reactiveui/ReactiveUI.Samples/tree/main/avalonia/suspension/
3. Launch `ReactiveUI.Samples.Suspension.sln`,
4. Try updating ReactiveUI to the latest version,
5. `RoutingState.NavigationStack` is no longer saved to disk when the app is closed.

### Reproduction repository

https://github.com/reactiveui/ReactiveUI.Samples/

### Expected behavior

ReactiveUI suspension feature used to be working according to this guide: https://habr.com/en/post/462307/
And according to the documentation page: https://www.reactiveui.net/docs/handbook/data-persistence/
Probably worth marking `NavigationStack` as `DataMember` again and adding back a default ctor, if this won't cause issues, and if these were removed unintentionally.

### ReactiveUI Version

17.0.1+

### PR DESCRIPTION
**Pull request by tomasfil**
fixes #3322

### CODE REVIEW

**Comment by ChrisPulman** [comment]
> I am unsure if this build error is my fault or not

I will try to check it this evening to see what's happening, thank you

**Comment by ChrisPulman** [comment]
[Link to VS2022 Bug](https://developercommunity.visualstudio.com/t/XamarinMac-binaries-are-missing-in-173/10164443?entry=problem&ref=native&refTime=1665009130654&refUserId=f22990ed-1d2a-49af-9580-7af595a9d047)

**Review by ChrisPulman — APPROVED** [review]

**Comment by glennawatson** [comment]
Sooner they fix that problem so we can remove those ugly workarounds the better.

**Comment by ChrisPulman** [comment]
> Sooner they fix that problem so we can remove those ugly workarounds the better.

Agree, I will keep an eye on it

File: src/ReactiveUI/Routing/RoutingState.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public RoutingState(IScheduler? scheduler = null)
     /// Gets the current navigation stack, the last element in the
     /// collection being the currently visible ViewModel.
     /// </summary>
-    [IgnoreDataMember]
+    [DataMember]
     public ObservableCollection<IRoutableViewModel> NavigationStack { get; }
```
Comment (worldbeater) [inline src/ReactiveUI/Routing/RoutingState.cs:51]: Not sure if just changing `[IgnoreDataMember]` to `[DataMember]` fixes the issue #3322. According to L42, the `NavigationStack` property is only initialized inside the constructor and offers no public setter for a serializer. https://github.com/reactiveui/ReactiveUI/blob/2e3b6000f6cc9a695617dfe2647f87a2f4057044/src/ReactiveUI/Routing/RoutingState.cs#L42

**Pull request merged**